src/common/helpers/process-case-with-workflow.js

```diff
@@ -0,0 +1,79 @@
+export const processCaseWithWorkflow = async (selectedCase, workflow) => {
+  // Add titles from workflow stages to selectedCase stages
+  selectedCase.stages = selectedCase.stages.map((stage) => {
+    const workflowStage = workflow.stages.find((ws) => ws.id === stage.id)
+
+    // Add title from workflow to the stage
+    const updatedStage = {
+      ...stage,
+      title: workflowStage?.title
+    }
+
+    // Add titles to task groups
+    if (stage.taskGroups && stage.taskGroups.length > 0) {
+      updatedStage.taskGroups = stage.taskGroups.map((taskGroup) => {
+        const workflowTaskGroup = workflowStage?.taskGroups?.find(
+          (wtg) => wtg.id === taskGroup.id
+        )
+
+        // Add title to task group
+        const updatedTaskGroup = {
+          ...taskGroup,
+          title: workflowTaskGroup?.title
+        }
+
+        // Add titles to tasks
+        if (taskGroup.tasks && taskGroup.tasks.length > 0) {
+          updatedTaskGroup.tasks = taskGroup.tasks.map((task) => {
+            const workflowTask = workflowTaskGroup?.tasks?.find(
+              (wt) => wt.id === task.id
+            )
+
+            // Add title to task
+            return {
+              ...task,
+              title: workflowTask?.title,
+              type: workflowTask?.type
+            }
+          })
+        }
+
+        return updatedTaskGroup
+      })
+    }
+
+    if (workflowStage.actions) {
+      updatedStage.actions = workflowStage.actions
+    }
+
+    return updatedStage
+  })
```

Comment:
This will need a refactor to move combining cases and workflows to the API. We can do it after this stuff is over the line